lib/Dialect/Numpy/IR/NumpyOps.cpp

```diff
@@ -77,6 +77,49 @@ bool StaticInfoCastOp::areCastCompatible(mlir::TypeRange inputs,
          output.getDtype().isa<AnyDtypeType>();
 }
 
+void StaticInfoCastOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                   MLIRContext *context) {
+  // static_info_cast(oneUse@create_array_from_tensor(%tensor))
+  // -->
+  // create_array_from_tensor(tensor_static_info_cast(%tensor))
+  //
+  // This pattern tends to create more tensor code and less array code.
+  // This form is considered more canonical because it has same number of ops
+  // but is more analyzable.
+  //
+  // TODO: Consider a world where we numpy.ndarray can track an "immutable" bit
+  // which makes it tensor-like. Is that useful?
+  patterns.add(+[](StaticInfoCastOp op, PatternRewriter &rewriter) {
```

Comment:
Do you need the unary '+' in this case? (I think I recall that it becomes optional in newer standards)

Reply:
Needed on my machine sadly.